ports/raspberrypi/common-hal/alarm/__init__.c

```diff
@@ -234,6 +234,9 @@ void NORETURN common_hal_alarm_enter_deep_sleep(void) {
 
     // Reset uses the watchdog. Use scratch registers to store wake reason
     watchdog_hw->scratch[RP_WKUP_SCRATCH_REG] = _get_wakeup_cause();
+
+    // Just before reset, enable the pinalarm interrupt.
+    alarm_pin_pinalarm_entering_deep_sleep();
```

Comment:
I'm guessing there's a reason why not, but can interrupts be disabled or masked here to eliminate the rest of the race window?

Reply:
That routine sets a single boolean, so it's very tiny, and probably done inline, then the reset happens. Since the reset doesn't mess with the interrupts, even if I guarded `alarm_pin_pinalarm_entering_deep_sleep();`, there would be a time immediately after it, and before the reset_`cpu();`, when an interrupt might happen. So I'm not sure it would help any. But maybe I'm thinking about it wrong.

Reply:
Without reading the datasheet, I was imagining that you could disable interrupts, set up the wake interrupt sources, and then reset without ever enabling interrupts, a la:
```
disable_interrupt()
enable_pinalarm()
// (no enable_interrupt() here
reset()
```
this would make it so that there's no window before reset where interrupts are enabled. But it depends on the post-"reset" CPU state not caring that interrupts were disabled pre-"reset". This is analogous how __WFI for entering light sleep works in some ports, but I notice not raspberrypi (#5341, subsequently reverted by #5356)

Reply:
`reset_cpu()` is not actually a regular reset. This is its implementation:
https://github.com/adafruit/circuitpython/blob/f76351d17809f43afd6a77245d546951c7474050/ports/raspberrypi/supervisor/port.c#L203-L210